tests/integration/helpers.py

```diff
@@ -132,3 +134,89 @@ def check_jaas_config(model_full_name: str, unit: str):
             user_lines[matched[1]] = matched[2]
 
     return user_lines
+
+
+def _get_show_unit_json(model_full_name: str, unit: str) -> Dict:
+    """Retrieve the show-unit result in json format."""
+    show_unit_res = check_output(
+        f"JUJU_MODEL={model_full_name} juju show-unit {unit} --format json",
+        stderr=PIPE,
+        shell=True,
+        universal_newlines=True,
+    )
+
+    try:
+        show_unit_res_dict = json.loads(show_unit_res)
+        return show_unit_res_dict
+    except json.JSONDecodeError:
+        raise ValueError
+
+
+def check_acl_permission(host: str, password: str, folder: str, username: str = "super") -> None:
+    """Checks the existence of ACL permission of a given folder."""
+    kc = KazooClient(
+        hosts=host,
+        timeout=30.0,
+        sasl_options={"mechanism": "DIGEST-MD5", "username": username, "password": f"{password}"},
+    )
+
+    kc.start()
+    try:
+        value, _ = kc.get_acls_async(f"/{folder}") or None, None
+        stored_value = None
+        if value:
+            stored_value = value.get()
+        if stored_value:
+            assert stored_value is not None
+            return
+    except NoNodeError:
+        raise Exception("No ACL permission found!")
+    finally:
+        kc.stop()
+        kc.close()
+
+
+def get_relation_id(model_full_name: str, unit: str, app_name: str):
+    show_unit = _get_show_unit_json(model_full_name=model_full_name, unit=unit)
+    d_relations = show_unit[unit]["relation-info"]
+    for relation in d_relations:
+        if relation["endpoint"] == app_name:
+            relation_id = relation["relation-id"]
+            return relation_id
+    raise Exception("No relation found!")
+
+
+def get_relation_username(model_full_name: str, unit: str, app_name: str):
+    show_unit = _get_show_unit_json(model_full_name=model_full_name, unit=unit)
+    d_relations = show_unit[unit]["relation-info"]
+    for relation in d_relations:
+        if relation["endpoint"] == app_name:
+            return relation["application-data"]["username"]
+    raise Exception("No relation found!")
+
+
+def get_relation_chroot(model_full_name: str, unit: str, app_name: str):
+    show_unit = _get_show_unit_json(model_full_name=model_full_name, unit=unit)
+    d_relations = show_unit[unit]["relation-info"]
+    for relation in d_relations:
+        if relation["endpoint"] == app_name:
+            return relation["application-data"]["chroot"]
+    raise Exception("No relation found!")
+
+
+def get_relation_password(model_full_name: str, unit: str, app_name: str):
+    show_unit = _get_show_unit_json(model_full_name=model_full_name, unit=unit)
+    d_relations = show_unit[unit]["relation-info"]
+    for relation in d_relations:
+        if relation["endpoint"] == app_name:
+            return relation["application-data"]["password"]
+    raise Exception("No relation found!")
```

Comment:
IMO these three methods can be conjoined into 1, that returns `relation["application-data"]`

Then we can just reference in the tests. For example:

```python
rel_data = get_relation_data(model_full_name, unit, app)

check_acl_permission(host, username=rel_data["username"], password=rel_data["password"])
```

Something to that effect.